File: NET/Libraries/Treehopper.Libraries/Sensors/Temperature/Mlx90615.cs
using Treehopper;
using Treehopper.Libraries;
using Treehopper.Libraries.Sensors.Temperature;
/// <summary>
///     MLX90615 non-contact IR thermopile temperature sensor
/// </summary>
public class Mlx90615 : Mlx90614
{
    /// <summary>
    ///     Construct a new MLX90615 attached to the given i2c port
    /// </summary>
    /// <param name="module"></param>
    public Mlx90615(I2C module) : base(module)
    {
        dev = new SMBusDevice(0x5B, module);
        Object = new TempRegister(dev, 0x27);
        Ambient = new TempRegister(dev, 0x26);
    }
}

File: Java/api/treehopper.libraries/src/main/java/io/treehopper/libraries/sensors/temperature/Mlx90615.java
package io.treehopper.libraries.sensors.temperature;

import io.treehopper.SMBusDevice;
import io.treehopper.interfaces.I2c;

/**
 * Melexis MLX90615 non-contact I2c temperature sensor
 */
public class Mlx90615 extends Mlx90614 {

    public Temperature ambient;
    public Temperature object;
    SMBusDevice dev;

    public Mlx90615(I2c module) {
        super(module);

        ambient = new TempRegister(this.dev, (byte) 0x26);
        object = new TempRegister(this.dev, (byte) 0x27);
    }

    public double getRawIrData() {
        return dev.readWordData((byte) 0x25);
    }
}
